p2p/types/runner.go

```diff
@@ -0,0 +1,6 @@
+package types
+
+type Runner interface {
+	Sink() chan<- Work
+	Done() <-chan uint
```

Comment:
`chan bool` vs `chan uint` vs `chan struct{}`. I honestly don't know what's the differentiating factor when it comes to choosing one instead of the other. Should research this later.

All I can say about `uint` is that I expect values to be positive (_although up to this moment I haven't made use of the actual sent values but rather just the event of receipt of new values_)